chapters/annotations.tex

```diff
@@ -2052,24 +2055,28 @@ \subsection{Protection of Classes}\label{protection-of-classes}
   Same as \lstinline!Access.packageText! and additionally the class, or part of the class, can be copied.
 \end{enumerate}
 
-The \lstinline!access! annotation holds for the respective class and all classes
-that are hierarchically on a lower level, unless overridden by a
-\lstinline!Protection! annotation with \lstinline!access!.
-Overriding \lstinline!access=Access.hide! or \lstinline!access=Access.packageDuplicate!
-has no effect.
+The \lstinline!access! annotation holds for the respective class and all classes that are hierarchically on a lower level, unless overridden by a \lstinline!Protection! annotation with \lstinline!access!.
+Overriding \lstinline!access=Access.hide! or \lstinline!access=Access.packageDuplicate! has no effect.
+
+Classes should not use other classes in ways that contradict this protection.
+Tools must ensure that protected contents are not shown, even if classes do not meet this requirement.
+
+\begin{nonnormative}
+For instance a class with \lstinline!Access.hide! should not be used in the diagram layer of a class with \lstinline!Access.diagram!, and there should not be hyperlinks to classes with \lstinline!Access.icon! (from classes with visible documentation).
+\end{nonnormative}
```

Comment:
The rule about what tools should do when the usage rule has been violated would also deserve some illustration.
```suggestion
\end{nonnormative}

\begin{example}
Consider the following invalid use of a class with \lstinline!Access.hide!:
\begin{lstlisting}[language=modelica]
block MySecret
  RealOutput y = time;
  annotation(Protection(access = Access.hide));
end MySecret;

model M
  MySecret mySecret;
  Integrator integrator;
equation
  connect(mySecret.y, integrator.u);
  annotation(Protection(access = Access.diagram));
end M;
\end{lstlisting}
In order to not reveal the existence of the class \lstinline!MySecret!, a tool may choose to show the diagram with both \lstinline!mySecret! and all connections to it removed.
(The tool could also choose to not show the diagram of \lstinline!M! at all.)
When simulating \lstinline!M!, the tool must not store \lstinline!mySecret.y!

Had \lstinline!M! had \lstinline!Access.packageDuplicate! instead, one way for a tool to not reveal the existence of \lstinline!MySecret! would be to refuse to show the class definition of \lstinline!M! at all.
\end{example}
```

Reply:
I believe it is important to show that M and MySecret are part of the same encrypted package, so enclose them in package P (no need to have a longer name). For the final sentence I would just completely remove it or reformulate it based on the other example:

If a user entered
```
model My
  P.MySecret a;
end My;
```
That should fail similarly as if `P.MySecret` did not exist, and especially the icon for `P.MySecret` should not be used.

-
Basically a class may be revealed in Modelica text, in documentation, in comments, or just guessed. As soon as we say that a hidden class is different from a non-existent class we open up a way to reveal their existence.

Thus trying to force tools to treat hidden classes in badly designed package specially we would force tools to reveal the existence of hidden classes in normal cases; that doesn't make sense.

Reply:
> That should fail similarly as if P.MySecret did not exist, and especially the icon for P.MySecret should not be used.

This makes sense.  I'll use it to reformulate the example.